lib/ansible/modules/storage/netapp/na_ontap_svm.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/python
+
+# (c) 2018, NetApp, Inc
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+
+module: na_ontap_svm
+
+short_description: Manage NetApp Ontap svm
+extends_documentation_fragment:
+    - netapp.na_ontap
+version_added: '2.6'
+author: Sumit Kumar (sumit4@netapp.com), Archana Ganesan (garchana@netapp.com)
+
+description:
+- Create, modify or delete svm on NetApp Ontap
+
+options:
+
+  state:
+    description:
+    - Whether the specified SVM should exist or not.
+    choices: ['present', 'absent']
+    default: 'present'
+
+  name:
+    description:
+    - The name of the SVM to manage.
+    required: true
+
+  new_name:
+    description:
+    - New name of the SVM to be renamed
+    required: false
+
+  root_volume:
+    description:
+    - Root volume of the SVM. Required when C(state=present).
+
+  root_volume_aggregate:
+    description:
+    - The aggregate on which the root volume will be created.
+    - Required when C(state=present).
+
+  root_volume_security_style:
+    description:
+    -   Security Style of the root volume.
+    -   When specified as part of the vserver-create,
+        this field represents the security style for the Vserver root volume.
+    -   When specified as part of vserver-get-iter call,
+        this will return the list of matching Vservers.
+    -   Possible values are 'unix', 'ntfs', 'mixed'.
+    -   The 'unified' security style, which applies only to Infinite Volumes,
+        cannot be applied to a Vserver's root volume.
+    -   Valid options are "unix" for NFS, "ntfs" for CIFS,
+        "mixed" for Mixed, "unified" for Unified.
+    -   Required when C(state=present)
+    choices: ['unix', 'ntfs', 'mixed', 'unified']
+
+  allowed_protocols:
+    description:
+    - Allowed Protocols.
+    - When specified as part of a vserver-create,
+      this field represent the list of protocols allowed on the Vserver.
+    - When part of vserver-get-iter call,
+      this will return the list of Vservers
+      which have any of the protocols specified
+      as part of the allowed-protocols.
+    - When part of vserver-modify,
```

Comment:
this should probably be reworded without using API-isms- it's basically the thing discussed before on other list types, where you're accepting the absolute list, not a set to be merged with the existing list.